src/LatteContext/Collector/TemplateRenderCollector.php

```diff
@@ -137,7 +137,11 @@ public function processNodeInPhp(Node $node, Scope $scope): ?array
         }
 
         $calledType = $scope->getType($node->var);
-        if (!$this->templateTypeResolver->resolve($calledType)) {
+        if ($calledType instanceof ThisType) {
+            $calledType = $calledType->getStaticObjectType();
+        }
+        $engineType = new ObjectType('Latte\Engine');
+        if (!$this->templateTypeResolver->resolve($calledType) && $engineType->isSuperTypeOf($calledType)->no()) {
```

Comment:
->no() means it is maybe, I prefer to use `! ->yes()`

Reply:
I think we want to cover maybes too. If render is called on union type that includes `Latte\Engine`

Reply:
I think we should be consistent, so for now use ! yes as it is everywhere

Reply:
ok